server/mux.go

```diff
@@ -0,0 +1,115 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+
+	"github.com/soheilhy/cmux"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/reflection"
+)
+
+type MuxServer struct {
+	config Config
+	GRPCServer
+	HTTPServer
+}
+
+type MuxOption func(*muxOptions)
+
+type muxOptions struct {
+	grpcOptions
+	httpOptions
+}
+
+func WithMuxGRPCServerOptions(opts ...grpc.ServerOption) MuxOption {
+	return func(mos *muxOptions) {
+		WithGRPCServerOptions(opts...)(&mos.grpcOptions)
+	}
+}
+
+func WithMuxGRPCServer(grpcServer *grpc.Server) MuxOption {
+	return func(mos *muxOptions) {
+		WithGRPCServer(grpcServer)(&mos.grpcOptions)
+	}
+}
+
+func WithMuxHTTPServer(httpServer *http.Server) MuxOption {
+	return func(mos *muxOptions) {
+		WithHTTPServer(httpServer)(&mos.httpOptions)
+	}
+}
+
+func NewMux(config Config, options ...MuxOption) (*MuxServer, error) {
```

Comment:
why is cmux being added? we can just use different port for grpc & http

Reply:
The package provides an option for both use separate ports, or use on the same port. The reason allowing same port is that it enables users to be able to use both grpc and http without much changes in there infrastructure and deployments.